### PR TITLE
apply_role: Add progress logging

### PR DESCRIPTION
**Pull request by rsalevsky**
At the moment there is no easy way to watch the progress of applying a
proposal. This change introduces a new logger which logs progress
relevant messages to a seperate log. But also adds this messages as
"PROGRESS" messages to the production log.

Interesting for https://github.com/sap-oc/crowbar-core/issues/6

### CODE REVIEW

File: crowbar_framework/config/initializers/progress_logger.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    log_file = if Rails.env.production?
+      File.join(ENV["CROWBAR_LOG_DIR"], "progress.log")
+    else
+      Rails.root.join("log", "progress.log")
+    end
+
+    logger = Logger.new(log_file)
+    logger.formatter = ::Crowbar::Logger::Formatter.new
+    logger.info(progname)
```
Comment (Itxaka) [inline crowbar_framework/config/initializers/progress_logger.rb]: Is this part necessary? Seems like we are logging twice no? Other logger levels do nothing like this so could you clarify why is this needed?

Comment (Itxaka) [inline crowbar_framework/config/initializers/progress_logger.rb]: oh, I see, this goes to a separate file + our normal file :+1:

Comment (rsalevsky) [inline crowbar_framework/config/initializers/progress_logger.rb]: This part is necessary, but the `add(idx, nil, progname, &block)` could be worth discussing. To log the progress to a separate log is direct customer feedback to allow to track the progress in a easy way.

Comment (Itxaka) [inline crowbar_framework/config/initializers/progress_logger.rb]: for me this is great, having on a separate file as per customer feedback + having it on the normal log file to have the full workflow in the same place, thanks!
